### PR TITLE
More `db.DefaultContext` refactor

### PR DESCRIPTION
**Pull request by JakobDev**
Part of #27065

This PR touches functions used in templates. As templates are not static typed, errors are harder to find, but I hope I catch it all. I think some tests from other persons do not hurt.

### CODE REVIEW

File: templates/projects/new.tmpl [hunk 1]
```diff
@@ -29,7 +29,7 @@
 					<div class="default text">{{ctx.Locale.Tr "repo.projects.template.desc_helper"}}</div>
 					<div class="menu">
 						{{range $element := .BoardTypes}}
-							<div class="item" data-id="{{$element.BoardType}}" data-value="{{$element.BoardType}}">{{ctx.Locale.Tr ctx.Locale.Translation}}</div>
+							<div class="item" data-id="{{$element.BoardType}}" data-value="{{$element.BoardType}}">{{ctx.Locale.Tr $element.Translation}}</div>
```
Comment (delvh) [inline templates/projects/new.tmpl]: oops.

**Review by delvh — APPROVED** [review]

File: templates/projects/new.tmpl (hunk 1 above)
Comment (wxiaoguang) [reply templates/projects/new.tmpl]: I would suggest to fix these bugs in a separate PR ASAP.

(Well, that's why I said I do not feel your regexp is safe .....)

Comment (JakobDev) [reply templates/projects/new.tmpl]: The big problem here is, that templates are not static typed, so it's easy to miss those errors. We really need some frontend tests. I think it's enough to load the page and see if it throws an error.

I needed to add a project to test things in this PR and it landed in git. If you want to create a separate PR to fix this, you are free to do this.

Comment (delvh) [reply templates/projects/new.tmpl]: #27275

**Comment by lunny** [comment]
```log
Render failed, failed to render template: user/dashboard/dashboard, error: template error: builtin(static):user/dashboard/feeds:84:20 : executing "user/dashboard/feeds" at <.GetRepoLink>: wrong number of args for GetRepoLink: want 1 got 0
----------------------------------------------------------------------
					{{$repoLink := .GetRepoLink}}
					               ^
----------------------------------------------------------------------
```

**Comment by delvh** [comment]
ping for reviewers.

**Review by lunny — APPROVED** [review]

**Pull request merged**